server/src/test/java/org/opensearch/rest/RestControllerTests.java

```diff
@@ -638,6 +643,42 @@ public Exception getInboundException() {
         );
     }
 
+    // Tests to check authenticate(...) method
+    public void testRestRequestAuthenticationSuccess() {
+        final AuthenticationManager authManager = new InternalAuthenticationManager();
+        Identity.setAuthManager(authManager);
+
+        final ThreadContext threadContext = client.threadPool().getThreadContext();
+
+        final FakeRestRequest fakeRestRequest = new FakeRestRequest.Builder(NamedXContentRegistry.EMPTY).withHeaders(
+            Collections.singletonMap("Authorization", Collections.singletonList("Basic YWRtaW46YWRtaW4="))
+        ) // admin:admin
+            .build();
+        final AssertingChannel channel = new AssertingChannel(fakeRestRequest, true, RestStatus.OK);
+        restController.dispatchRequest(fakeRestRequest, channel, threadContext);
+
+        assertTrue(channel.getSendResponseCalled());
+    }
+
+    public void testRestRequestAuthenticationFailure() {
+        final AuthenticationManager authManager = new InternalAuthenticationManager();
+        Identity.setAuthManager(authManager);
+
+        final ThreadContext threadContext = client.threadPool().getThreadContext();
+
+        final FakeRestRequest fakeRestRequest = new FakeRestRequest.Builder(NamedXContentRegistry.EMPTY).withHeaders(
+            Collections.singletonMap("Authorization", Collections.singletonList("Basic bWFydmluOmdhbGF4eQ=="))
+        ) // marvin:galaxy
+            .build();
+
+        // RestStatus is OK even though the authn information is incorrect. This is because we, yet, don't fail the request
+        // if it was unauthorized. The status should be changed to UNAUTHORIZED once the flow is updated.
```

Comment:
Before we merge I would love to see a test run the returns a 401 because invalid credentials we used

Reply:
Afaik, We decided to not fail the unauthorized requests, but if that has changed I can work on adding that piece of code here.

https://github.com/opensearch-project/OpenSearch/pull/4798/files#diff-10e11897b4dcd1403f4a484fe323f8c7f2bdd180a6277f5ec3d24fc0b1d6c168R633-R642

Reply:
What is our alternative if we are not failing unauthorized requests? I assume we are still not fulfilling them but what sort of response are we sending if not a 401?

Reply:
For now we let the request flow its natural course as if there was no authentication

Reply:
Maybe this is a good time to add a feature flag in `opensearch.yml`? If the feature flag is enabled then we should not pass through the request and instead reject it.

Reply:
IMO this should be added as a feature unless we want to add the whole auth functionality as a toggle-able feature. Adding just this feature will be a coding overhead and we might not need once we finalize on how to proceed with auth failure